plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/generator/SQLGeneratorDialog.java

```diff
@@ -278,16 +260,27 @@ public void widgetSelected(SelectionEvent e) {
                 public void widgetSelected(SelectionEvent e) {
                     sqlGenerator.setShowCastParams(supportsCastParamsButton.getSelection());
                     getDialogBoundsSettings().put(DBPScriptObject.OPTION_CAST_PARAMS, supportsCastParamsButton.getSelection());
-
-                    UIUtils.runInUI(sqlGenerator);
-                    Object sql = sqlGenerator.getResult();
-                    if (sql != null) {
-                        setSQLText(CommonUtils.toString(sql));
-                        updateSQL();
-                    }
+                    startGenerateJob();
                 }
             });
         }
         return composite;
     }
+
+    private void startGenerateJob() {
+        generateDDLJob.cancel();
+        generateDDLJob.schedule();
```

Comment:
Why doi you schedule job before disabling button and updating SQL?
PrepareUI first, then start generating then enableUI.

Reply:
Fixed